pkg/apis/sources/v1alpha1/cloudevents_types.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2022 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/triggermesh/triggermesh/pkg/apis/common/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+)
+
+// +genclient
+// +genreconciler
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CloudEventsSource is the Schema for the event source.
+type CloudEventsSource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   CloudEventsSourceSpec `json:"spec,omitempty"`
+	Status v1alpha1.Status       `json:"status,omitempty"`
+}
+
+// Check the interfaces the event source should be implementing.
+var (
+	_ v1alpha1.Reconcilable = (*CloudEventsSource)(nil)
+	_ v1alpha1.EventSender  = (*CloudEventsSource)(nil)
+)
+
+// CloudEventsSourceSpec defines the desired state of the event source.
+type CloudEventsSourceSpec struct {
+	duckv1.SourceSpec `json:",inline"`
+
+	// Credentials to connect to this source.
+	// +optional
+	Credentials *HTTPCredentials `json:"credentials,omitempty"`
+
+	// Path under which request are accepted
+	// +optional
+	Path *string `json:"path,omitempty"`
+
+	// RateLimiter for incoming events per adapter instance.
+	// A single CloudEventsSource object can create multiple adapter instances,
+	// the rate limiting configuration being applied to each of them individually.
+	// +optional
+	RateLimiter *RateLimiter `json:"rateLimiter,omitempty"`
+}
+
+// HTTPCredentials to be used when receiving requests.
+type HTTPCredentials struct {
+	BasicAuths []HTTPBasicAuth `json:"basicAuths,omitempty"`
+	Tokens     []HTTPToken     `json:"tokens,omitempty"`
+}
+
+// HTTPBasicAuth credentials.
+type HTTPBasicAuth struct {
+	Username string                  `json:"username"`
+	Password v1alpha1.ValueFromField `json:"password"`
+}
+
+// HTTPToken credentials.
+type HTTPToken struct {
+	Header string                  `json:"header"`
```

Comment:
Wouldn't the header always be `Authorization` for token based auth?

Reply:
It could be!

I have been hesitant and left it open because I could not find the proper RFC for a token scheme:

- [Basic](https://datatracker.ietf.org/doc/html/rfc7617) seems to be only for user/password
- [Bearer](https://datatracker.ietf.org/doc/html/rfc6750) seems to be only for OAuth 2.0

This is not OAuth 2.0, but you can find services around that expect
```
Authorization: Bearer t00k33n123456
```

Is it ok to use an static token with `Authorization: Bearer`? 
I am not sure. Maybe I should remove the token authentication.

WDYT?

Reply:
@cab105 I will be removing the token based authentication.

we can iterate later to add JWT if needed (or some other authentication from an existing spec), but this seems very ad-hoc.
Thanks for pointing this.